src/app/detail/ui/ArchiveWrapper.tsx

```diff
@@ -0,0 +1,66 @@
+'use client'
+
+import { ArchiveBtn } from '@/components/buttons/ArchiveBtn'
+import { useEffect, useState } from 'react'
+import useSWR, { useSWRConfig } from 'swr'
+import { fetcher } from '@/utils/fetcher'
+import { useSelector } from 'react-redux'
+import { selectTaskBoard } from '@/redux/features/taskBoardSlice'
+import { Skeleton } from '@mui/material'
+import store from '@/redux/store'
+import { setTask } from '@/redux/features/taskDetailsSlice'
+
+export const ArchiveWrapper = ({ taskId }: { taskId: string }) => {
+  const { token } = useSelector(selectTaskBoard)
+  const { mutate } = useSWRConfig()
+  const cacheKey = `/api/tasks/${taskId}?token=${token}`
+  const { data, error } = useSWR(cacheKey, fetcher)
+  const [isArchived, setIsArchived] = useState<boolean | undefined>(undefined)
+
+  // Set the initial state when `data` becomes available
+  useEffect(() => {
+    if (data?.task) {
+      setIsArchived(data.task.isArchived)
+      store.dispatch(setTask(data.task))
+    }
+  }, [data])
```

Comment:
I don't think this is listening for realtime archive status updates

Reply:
Added real time in the commit below